crates/core/src/key/common.rs

```diff
@@ -22,6 +24,16 @@ use crate::ethereum_events::EthAddress;
 use crate::key::{SignableBytes, StorageHasher};
 use crate::{impl_display_and_from_str_via_format, string_encoding};
 
+#[derive(BorshSchema)]
+enum CommonPublicKey {
+    /// Encapsulate Ed25519 public keys
+    #[allow(unused)]
+    Ed25519(ed25519::PublicKey),
+    /// Encapsulate Secp256k1 public keys
+    #[allow(unused)]
+    Secp256k1(secp256k1::PublicKey),
+}
```

Comment:
hm I guess the only reason you created this enum is to avoid sharing a `PublicKey` alias with other public key types in the borsh schema definitions. however, this is an error prone and hacky approach IMO. I'd suggest either doing something like this:
```suggestion
#[derive(BorshSchema)]
#[repr(transparent)]
struct CommonPublicKey(PublicKey);
```
or, even better, using the attributes available to the `BorshSchema` derive macro: https://docs.rs/borsh/latest/borsh/derive.BorshSchema.html#4-borshschemaparams---field-level-attribute

Reply:
nvm just realized the first solution wouldn't work :P we'd be using a recursive definition

Reply:
Agreed - this was hacky and error prone. I just looked at the `BorshSchema` derive macro attributes. Probably I'm missing something, but the relevant attributes all seem to be at field level. So while it's possible to manipulate the schemas of the subcomponents of a `PublicKey`, it doesn't seem to be possible to modify, say, the declaration of a `PublicKey` itself. No?

That being said, I just pushed an approach of modifying the declaration alias without duplicating the `PublicKey` `struct`. The new approach just uses a type alias instead. Would this be acceptable?

Reply:
> That being said, I just pushed an approach of modifying the declaration alias without duplicating the PublicKey struct. The new approach just uses a type alias instead. Would this be acceptable?

I think this is fine, yeah! We had a similar approach while developing the Ethereum Bridge, to avoid name collisions w/ borsh schemas.